contrib/auto-render/test/auto-render-spec.js

```diff
@@ -234,3 +235,19 @@ describe("A delimiter splitter", function() {
             ]);
     });
 });
+
+describe("Pre-process callback", function() {
+    it("replace `²` with `^2 `", function() {
+        const el1 = document.createElement('div');
+        el1.textContent = 'Circle equation: $x² + y² = r²$.';
+        const el2 = document.createElement('div');
+        el2.textContent = 'Circle equation: $x^2 + y^2 = r^2$.';
+        const delimiters = [{left: "$", right: "$", display: false}];
+        renderMathInElement(el1, {
+            delimiters,
+            preProcess: math => math.replace(/²/g, '^2'),
```

Comment:
Does the test fail without this?  I know we handle some unicode chars, but I forget if we handle superscript chars properly without the preprocessor you've defined here.  It might be worth writing a more extreme preprocessor such as `/-squared/g, '^2'` and then run it on `x-squared + y-squared = r-squared`.  The rationale is that even if we don't handle superscript chars right now, we might at some point in the future which would mean that this test would loose its power at that point.

Reply:
Yes the test fails without `preProcess`, however I understand your concern. So I update the test in order to be more future-proof.